backend/src/connector/mod.rs

```diff
@@ -1 +1,328 @@
+    /// connectors.
+    #[must_use]
+    pub const fn builder() -> Builder {
+        Builder {}
+    }
+}
+
+/// Builder for HTTP(S) connectors
+#[derive(Debug)]
+pub struct Builder {}
+
+impl Builder {
+    /// [Stub] Use HTTPS instead of HTTP
+    #[must_use]
+    pub const fn https(self) -> HttpsBuilder {
+        HttpsBuilder {}
+    }
+
+    /// Build a HTTP connector
+    #[must_use]
+    pub fn build(self) -> hyper::client::connect::HttpConnector {
+        hyper::client::connect::HttpConnector::new()
+    }
+}
+
+// TODO
+/// [Stub] Builder for HTTPS connectors
+#[derive(Debug)]
+pub struct HttpsBuilder {}
+
+impl HttpsBuilder {
+    pub fn build(self) {
+        unimplemented!()
+    }
+}
+
+#[derive(Clone)]
+pub struct BobClient<Client: ApiNoContext<ClientContext> + Send + Sync> {
+    /// Unique Identifier
+    id: Uuid,
+
+    /// Bob's hostname
+    hostname: Hostname,
+
+    // NOTE: Can (and should) the API interface mutate?..
+    /// Connection,
+    main: Arc<Client>,
+
+    /// Clients for all known nodes
+    cluster: HashMap<NodeName, Arc<Client>>,
+}
+
+#[allow(clippy::missing_fields_in_debug)]
+impl<Client: ApiNoContext<ClientContext> + Send + Sync + Clone> std::fmt::Debug
+    for BobClient<Client>
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let user = &self
+            .main
+            .context()
+            .auth_data
+            .as_ref()
+            .map_or("Unknown", |cred| cred.username());
+        f.debug_struct("BobClient")
+            .field("hostname", &self.hostname)
+            .field("user", &user)
+            .finish()
+    }
+}
+
+impl<ApiInterface: ApiNoContext<ClientContext> + Send + Sync> BobClient<ApiInterface> {
+    // impl BobClient<C> {
+    /// Creates new [`BobClient`] from [`BobConnectionData`]
+    ///
+    /// # Errors
+    /// The function will fail if a hostname isn't a valid url or the client couldn't establish
+    /// connection for the BOB cluster
+    pub async fn try_new(
+        bob_data: BobConnectionData,
+        timeout: RequestTimeout,
+    ) -> Result<HttpBobClient, ClientError> {
+        let auth = bob_data
+            .credentials
+            .map(|creds| Authorization::basic(&creds.login, &creds.password));
+        let hostname = bob_data.hostname.clone();
+
+        let context: ClientContext = ClientContext {
+            timeout,
+            auth_data: auth,
+            xspan: XSpanIdString::default(),
+        };
+        let client =
+            Client::try_new_http(&hostname.to_string()).change_context(ClientError::InitClient)?;
+        let nodes = client
+            .clone()
+            .with_context(context.clone())
+            .get_nodes()
+            .await
+            .change_context(ClientError::Inaccessible)?;
+        let api::GetNodesResponse::AJSONArrayOfNodesInfoAndVdisksOnThem(nodes) = nodes else {
+            Err(APIError::InvalidStatusCode(StatusCode::FORBIDDEN))
```

Comment:
Why do you think that any error here is `Forbidden`? It seems that this is not very good and it is better to use a real error as the base one

Reply:
there is no other response from Bob API, according to [his OpenAPI scheme](https://github.com/qoollo/bob/blob/519c9501786548a0470af714f7a3c0762dab89a7/config-examples/openapi.yaml#L77)

But that's still bad to assume that's no other error can't be received.
Now all responses can be turned to corresponding status codes